updates/upgrader.go

```diff
@@ -113,38 +113,62 @@ func upgradePortmasterControl() error {
 		return nil
 	}
 
-	// update portmaster-control in data root
-	rootControlPath := filepath.Join(filepath.Dir(registry.StorageDir().Path), filename)
-	err := upgradeFile(rootControlPath, pmCtrlUpdate)
+	// update portmaster-start in data root
+	rootPmStartPath := filepath.Join(filepath.Dir(registry.StorageDir().Path), filename)
+	err := upgradeFile(rootPmStartPath, pmCtrlUpdate)
 	if err != nil {
 		return err
 	}
-	log.Infof("updates: upgraded %s", rootControlPath)
+	log.Infof("updates: upgraded %s", rootPmStartPath)
 
-	// upgrade parent process, if it's portmaster-control
+	// TODO(ppacher): remove once we released a few more versions ...
+	warnOnIncorrectParentPath(filename)
+
+	return nil
+}
+
+func warnOnIncorrectParentPath(expectedFileName string) {
+	// upgrade parent process, if it's portmaster-start
 	parent, err := processInfo.NewProcess(int32(os.Getppid()))
 	if err != nil {
-		return fmt.Errorf("could not get parent process for upgrade checks: %w", err)
+		log.Tracef("could not get parent process: %s", err)
+		return
 	}
 	parentName, err := parent.Name()
 	if err != nil {
-		return fmt.Errorf("could not get parent process name for upgrade checks: %w", err)
+		log.Tracef("could not get parent process name: %s", err)
+		return
 	}
-	if parentName != filename {
-		log.Tracef("updates: parent process does not seem to be portmaster-control, name is %s", parentName)
-		return nil
+	if parentName != expectedFileName {
+		log.Warningf("updates: parent process does not seem to be portmaster-start, name is %s", parentName)
+
+		// TODO(ppacher): once we released a new installer and folks had time
+		//                to update we should send a module warning/hint to the
+		//                UI notifying the user that he's still using portmaster-control.
+		return
 	}
+
 	parentPath, err := parent.Exe()
 	if err != nil {
-		return fmt.Errorf("could not get parent process path for upgrade: %w", err)
+		log.Tracef("could not get parent process path: %s", err)
+		return
 	}
-	err = upgradeFile(parentPath, pmCtrlUpdate)
+
+	absPath, err := filepath.Abs(parentPath)
 	if err != nil {
-		return err
+		log.Tracef("could not get absolut parent process path: %s", err)
+		return
 	}
-	log.Infof("updates: upgraded %s", parentPath)
 
-	return nil
+	root := filepath.Dir(registry.StorageDir().Path)
+	if !strings.HasPrefix(absPath, root) {
+		log.Warningf("detected unexpected path %s for portmaster-start", absPath)
+
+		// TODO(ppacher): once we released a new installer and folks had time
+		//                to update we should send a module warning/hint to the
+		//                UI notifying the user that he's using portmaster-start
+		//                from a wrong location.
```

Comment:
Can this happen at all if we make the switch to portmaster-start?

I'd rather not update it at all outside of the data dir and just give a warning to the user whenever the parent _is_ portmaster-start, but _is not_ in the data directory - forever.

Reply:
Got something mixed up. Nevermind.